pandas/io/excel/_util.py

```diff
@@ -224,7 +274,9 @@ def fill_mi_header(row, control_row):
     return row, control_row
 
 
-def pop_header_name(row, index_col):
+def pop_header_name(
+    row: list[Hashable], index_col: int | list[int]
```

Comment:
Could also try to use TypeVar here to link the `list[Hashable]`. Probably only useful if the code later requires a more precise type.

Reply:
Not sure if this is useful here